test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Internal/DefaultPageHandlerMethodSelectorTest.cs

```diff
@@ -235,6 +234,102 @@ public void Select_ReturnsHandlerThatMatchesFormAction()
         }
 
         [Fact]
+        public void Select_FormActionFromQueryString()
+        {
+            // Arrange
+            var descriptor1 = new HandlerMethodDescriptor
+            {
+                HttpMethod = "POST",
+                FormAction = new StringSegment("Add"),
+            };
+
+            var descriptor2 = new HandlerMethodDescriptor
+            {
+                HttpMethod = "POST",
+                FormAction = new StringSegment("Delete"),
+            };
+
+            var pageContext = new PageContext
+            {
+                ActionDescriptor = new CompiledPageActionDescriptor
+                {
+                    HandlerMethods =
+                    {
+                        descriptor1,
+                        descriptor2,
+                    },
+                },
+                RouteData = new RouteData(),
+                HttpContext = new DefaultHttpContext
+                {
+                    Request =
+                    {
+                        Method = "Post",
+                        QueryString = new QueryString("?formaction=Delete"),
+                    },
+                },
+            };
+            var selector = new DefaultPageHandlerMethodSelector();
+
+            // Act
+            var actual = selector.Select(pageContext);
+
+            // Assert
+            Assert.Same(descriptor2, actual);
+        }
+
+        [Fact]
+        public void Select_FormActionConsidersRouteDataFirst()
```

Comment:
I opted to just use the first item. If we left it as it was in the case that we had 2 it would have resulted in `FirstOne,SecondOne`.